common/quotas/global.go

```diff
@@ -41,8 +41,3 @@ func PerMember(service string, globalRPS, instanceRPS float64, resolver membersh
 	avgQuota := math.Max(globalRPS/float64(memberCount), 1)
 	return math.Min(avgQuota, instanceRPS)
 }
-
-// PerMemberDynamic is a dynamic variant (using RPSFunc) of PerMember
-func PerMemberDynamic(service string, globalRPS, instanceRPS RPSFunc, resolver membership.Resolver) RPSFunc {
```

Comment:
I'm not sure I understand this change. PerMemberDynamic is removed, though you've introduced much more closures down the code. It looks like it is now more code/harder to read, though easier to understand.

Reply:
This function takes two functions, then returns a function which calls a function with these two functions as input. Additionally, all the callers need to "convert" their functions to match the signatures of this function. For this they use the .AsFloat64 method, which is defined on the function type. 
The function is (IMO) extremely convoluted, which is the reason I removed it. 

Now the callers just pass a simple closure, and thats it.

This is not the important change so we can take it out for now, the other change (your other comment) is the important one. But I vote to keep this change as well 🙂